com.sap.cloud.lm.sl.cf.process/src/test/java/com/sap/cloud/lm/sl/cf/process/util/ApplicationArchiveExtractorTest.java

```diff
@@ -0,0 +1,101 @@
+            Arguments.of(SAMPLE_MTAR_WITH_JAR_ENTRY_ABSOLUTE_PATH, "/web/", MessageFormat.format(FileUtils.PATH_SHOULD_NOT_BE_ABSOLUTE, "/web/"), MAX_UPLOAD_FILE_SIZE),
+            Arguments.of(SAMPLE_MTAR_WITH_JAR_ENTRY_NOT_NORMALIZED_PATH, "web/", MessageFormat.format(FileUtils.PATH_SHOULD_BE_NORMALIZED, "web/../asd"), MAX_UPLOAD_FILE_SIZE),
+            Arguments.of(SAMPLE_MTAR, "db/", MessageFormat.format(ERROR_SIZE_OF_APP_EXCEEDS_MAX_SIZE_LIMIT, 200), 200l),
+            Arguments.of(SAMPLE_MTAR, "web/web-server.zip", MessageFormat.format(ERROR_SIZE_OF_APP_EXCEEDS_MAX_SIZE_LIMIT, 200), 200));
+        // @formatter:on
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    public void testExtract(String mtar, String fileName, String expectedException, long maxFileUploadSize) throws Exception {
+        extractor = getApplicationArchiveExtractor(mtar, fileName, maxFileUploadSize);
+        mockZipEntryValidation();
+        if (expectedException == null) {
+            appPath = extractor.extract();
+            assertTrue(Files.exists(appPath));
+            return;
+        }
+
+        Exception exception = Assertions.assertThrows(Exception.class, () -> appPath = extractor.extract());
+        assertEquals(expectedException, exception.getCause()
+            .getMessage());
+    }
+
+    private ApplicationArchiveExtractor getApplicationArchiveExtractor(String mtar, String fileName, long maxFileUploadSize) {
+        InputStream mtarInputStream = getClass().getResourceAsStream(mtar);
+        return Mockito.spy(new ApplicationArchiveExtractor(mtarInputStream, fileName, maxFileUploadSize, null) {
+            @Override
+            protected void cleanUp(Path appPath) {
+                if (appPath == null || !Files.exists(appPath)) {
+                    return;
+                }
+                org.apache.commons.io.FileUtils.deleteQuietly(appPath.toFile());
+            }
+        });
+    }
+
+    private void mockZipEntryValidation() {
+        Mockito.doAnswer(new Answer<Void>() {
+            @Override
+            public Void answer(InvocationOnMock invocation) throws Throwable {
+                ZipEntry entry = (ZipEntry) invocation.getArguments()[0];
+                String path = entry.getName();
+                if (!path.equals(FilenameUtils.normalize(path, true))) {
+                    throw new IllegalArgumentException(MessageFormat.format(FileUtils.PATH_SHOULD_BE_NORMALIZED, path));
+                }
+                if (FilenameUtils.getPrefixLength(path) != 0 || Paths.get(path)
+                    .isAbsolute()) {
+                    throw new IllegalArgumentException(MessageFormat.format(FileUtils.PATH_SHOULD_NOT_BE_ABSOLUTE, path));
+                }
+                return null;
+            }
+        })
+            .when(extractor)
+            .validateEntry(Mockito.any(ZipEntry.class));
+    }

```

Comment:
since the method is protected, you can just Override it above next to cleanUp and not use a mock at all. This in my opinion would make the code easier to read.